[43842] Re-design and small changes to modal dialogues (DS) for coherence

bsatarnejad opened the pull request:
Modal redesign changes:

- [x] WP deletion changes:

The header section does not have a divider.
The text has been modified to include the name of the WP inside.
The action button changed to “Delete” and now includes an icon.

- [x] Invite user:

Modal dialog doesn’t include “X” button on the header anymore.
The action bar now contains the option button “Back” and the always present option to “cancel”
The process has been simplified to only three steps instead of one step per each form field.
Now the form fields “Name or email address”, “Role in project [X]” and “Invitation message” are in a single step.
The guidelines texts in the “Role in project [X]” and “Invitation message” are now more generic (no user name).
The last step now also includes the text from the field “Invitation message”.

- [x] Log time:

Modal dialog doesn’t include “X” button on the header anymore.

- [x] Add widget:

Modal dialog doesn’t include “X” button on the header anymore.
The list in the middle doesn’t have dividers anymore as now is using the spot-list default component as base.
There is an action bar with only one action “Cancel”.

- [x] New Board:

Modal dialog doesn’t include “X” button on the header anymore.
There is an action bar with only one action “Cancel”.
The new width is 60REM (960px) as 70REM is removed as width size.

- [x] Board > Configure this board:

Modal dialog doesn’t include “X” button on the header anymore.
The size of the modal depends on the content and not a fix height.

- [x] Work package table > Table settings: 

Modal dialogs doesn’t include “X” button on the header anymore.
The height of the modal is now fixed depending on the highest tab option (in this case the tab “Display settings” in order to avoid flickering behavior. 
The action bar is now gray and the order of the buttons is with the positive action in the right. 
The new width is 60REM (960px) as 70REM is removed as width size.

- [x] Work package table > Export:

Modal dialogs doesn’t include “X” button on the header anymore.
There is an action bar with only one action “Cancel”.

- [x] Work package table/Calendar/Board/Team planner > Save as:

Modal dialog doesn’t include “X” button on the header anymore.
The divider in the content has been switched with the “Visibility settings” header.
“Visibility settings” header font is reduced.
The action bar now includes two buttons “Cancel” and “Save”. Only the ”Save” includes icon.

- [x] Work package table/Calendar/Board/Team planner > Visibility settings:

Modal dialogs doesn’t include “X” button on the header anymore.
The size of the modal depends on the content and not a fix height.
The action bar now includes two buttons “Cancel” and “Save”. Only the ”Save” includes icon.

- [x] Help text:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider.
The action bar now includes two buttons “Close” and “Edit”.

- [x] Help > Introduction video:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have color.
The action bar now includes two buttons“Close”.
The text ”You can restart this video from the help menu” is now part of the description.

- [x] My account > Two-factor authentication > Backup codes password:

Modal dialogs doesn’t include “X” button on the header anymore.
The action bar now includes two buttons“Cancel” and “Confirm”.

- [x] My account > Two-factor authentication > Delete authentication:

Modal dialogs doesn’t include “X” button on the header anymore. 
The action bar now includes two buttons“Cancel” and “Confirm”.

- [x] Administration > Attribute help texts > Preview help text:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider. 
The action bar now includes two buttons “Close” and “Edit”.

- [x] Administration > Enterprise Edition > Delete token:

Modal dialogs doesn’t include “X” button on the header anymore.
The header section does not have divider.
All texts has been actualized to make the action clearer.
The action bar now includes two buttons “Cancel” and “Delete”. The second one now includes icon.

- [x] Administration > GDPR > Review DPA: 

Modal dialogs doesn’t include “X” button on the header anymore.
Now there is an action bar with the button “Close”.

- [x] Administration > API and webhooks > webhooks

Change header and body specified font size

https://community.openproject.org/projects/openproject/work_packages/43842/activity

bsatarnejad committed:
wp deletion modal changes

bsatarnejad committed:
fix test fails

bsatarnejad committed:
invite user modal changes

bsatarnejad committed:
fix eslint issus and add cancel button to summary modal

bsatarnejad committed:
add widget modal changes

bsatarnejad committed:
new board modal changes

bsatarnejad committed:
Merge branch 'dev' into redesign-of-modal-dialogues

bsatarnejad committed:
form configuration modal changes and remove unnecessary class

bsatarnejad committed:
WP table export modal changes

Kharonus commented:
Should we already review this or wait, until all checkboxes are checked?

bsatarnejad commented:
> Should we already review this or wait, until all checkboxes are checked?

Hi @Kharonus , I am still working on the PR. Some of the changes are applied.

bsatarnejad committed:
WP save query modal changes

bsatarnejad committed:
visibility setting modal changes

bsatarnejad committed:
help text modal changes

bsatarnejad committed:
help video modal changes

bsatarnejad committed:
2FA password confimation modal changes

bsatarnejad committed:
Delete EE token modal changes

bsatarnejad committed:
merge dev into this branch

bsatarnejad committed:
fix eslint errors

bsatarnejad committed:
eslint does not check the unbound method

bsatarnejad committed:
minor changes to switch the buttons and icon

bsatarnejad committed:
fix specs failing for invite user modal changes

bsatarnejad committed:
Merge branch 'dev' into redesign-of-modal-dialogues

bsatarnejad committed:
use the correct translations and revert the previous merge

Commit:
Fix opAutofocus directive

The opAutofocus directive was not setting focus if used like a normal
`autofocus` HTML attribute:

```
<input opAutofocus />
```

The above example would not produce a focus, because counterintuitively
Angular treats the default input to the directive here as an empty
string instead of undefined. This causes the default value of `true` NOT
to be set. Instead, the empty string was treated like the falsy value
it is.

bsatarnejad committed:
add focus on invite user modal while opening

bsatarnejad committed:
fix eslint errors

bsatarnejad committed:
merge dev into this branch

bsatarnejad committed:
eslint errors fix

bsatarnejad committed:
add icon delete for deleting ee token

b12f commented on frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html:
```diff
@@ -1,22 +1,20 @@
 <div
-  class="spot-modal wp-table--configuration-modal loading-indicator--location"
+  class="spot-modal loading-indicator--location"
```
Does this even have a loading state? I think we can delete it

b12f commented on frontend/src/app/shared/components/modals/wp-destroy-modal/wp-destroy.modal.html:
```diff
@@ -74,10 +72,11 @@
-        (click)="confirmDeletion($event)"
-      ></button>
+        (click)="confirmDeletion($event)">
+        <op-icon icon-classes="button--icon icon-delete"></op-icon>
```
Icons can now just be a span:
```suggestion
        <span class="spot-icon spot-icon_delete"></span>
```

b12f replied:
might still need the `button--icon` class tho

Kharonus replied:
I added some overwrite rules to use simple spans of `spot-icon` within the "old" buttons, but this is not merged yet.
See [here](https://github.com/opf/openproject/commit/12ed963f5fca1f3196edc19f456ab6f81680e486#diff-2c53795f153acc4b1f50994a3333ddaf2f6ef82a8ceee970352cc74730af1fb9).

But without this kind of overwrites `spot-icon` spans does not work within the old buttons.

b12f commented on frontend/src/app/features/invite-user-modal/invite-user.component.html:
```diff
@@ -9,42 +9,29 @@
-  (save)="onMessageSave($event)"
-  (back)="goTo(Steps.Role)"
-  (close)="closeMe()"
-></op-ium-message>
```
The component files for this and for the role step still need to be deleted

b12f commented on frontend/src/app/features/invite-user-modal/principal/principal.component.ts:
```diff
@@ -54,31 +62,50 @@ export class PrincipalComponent implements OnInit {
+      label: this.I18n.t('js.invite_user_modal.message.label'),
+      description: ():string => this.I18n.t('js.invite_user_modal.message.description', {
+        principal: this.principal?.name,
+      }),
```
The help texts for role and message are currently broken as long as a principal has not been selected yet:


![20220919_13h40m34s_grim](https://user-images.githubusercontent.com/11312895/191009478-656aba36-1a91-4acd-955c-61032e337379.png)

bsatarnejad replied:
I made changes in en.yml file for the messages, maybe it will take more time or after merging, the translation for other languages like German will be updated by Crowdin. Is it correct? or Should I change them myself?

b12f replied:
Ah I see, you're totally right, I'm sorry!

b12f commented on frontend/src/app/features/invite-user-modal/principal/principal.component.html:
```diff
@@ -60,16 +60,69 @@
+        *ngIf="roleControl?.touched && roleControl?.invalid"
+        >
+          {{ text.role.required }}
+        </div>
```
Not sure why, but the errors for role and principal are not shown correctly. To test, just open the second page and press next without changing anything:

![20220919_13h42m22s_grim](https://user-images.githubusercontent.com/11312895/191009675-e86366d8-3855-4dea-9c1c-e2e3c5e56600.png)

b12f commented on frontend/src/app/shared/components/grids/widgets/add/add.modal.html:
```diff
@@ -25,4 +25,16 @@
       class="grid--addable-widget"
     ></div>
   </div>
+
```
The list above here should be a `spot-list` but it not yet

b12f commented on frontend/src/app/features/work-packages/components/wp-table/configuration-modal/wp-table-configuration.modal.html:
```diff
@@ -24,19 +24,19 @@
   </div>
```
In this modal, the whole modal scrolls instead of the tab contents. I would expect this to be the other way around:


![20220919_13h51m54s_grim](https://user-images.githubusercontent.com/11312895/191011241-c870e751-6784-4ed2-81db-524481cf9ccc.png)

b12f commented on frontend/src/app/shared/components/modals/save-modal/save-query.modal.html:
```diff
@@ -30,7 +30,8 @@
     <section class="form--section">
-      <h3 [textContent]="text.label_visibility_settings"></h3>
+      <div class="spot-divider"></div>
+      <p>{{ text.label_visibility_settings }}</p> 
```
This has lost all heading styles. h3 might have been much, but making it a paragraph makes it unclear that this is supposed to be a heading:

![20220919_13h53m48s_grim](https://user-images.githubusercontent.com/11312895/191011545-b2b9f322-0d04-4ba3-b68c-b8786b18216d.png)

b12f commented on app/views/onboarding/_onboarding_video_modal.html.erb:
```diff
@@ -46,9 +46,8 @@ See COPYRIGHT and LICENSE files for more details.
 
```
Honestly I also think we can remove the icon from this modal header, but maybe @marcalcobe and @psatyal have different opinions?

![20220919_13h59m37s_grim](https://user-images.githubusercontent.com/11312895/191012389-a86cf9c1-1581-4f96-8ece-825a02e3e84a.png)

bsatarnejad replied:
As we discussed it in a meeting with designers, we can keep the icon here.

b12f reviewed (changes requested):
A ton of great work, but there's some last "Feinschliff" to be done :)

bsatarnejad committed:
Merge branch 'dev' into redesign-of-modal-dialogues

bsatarnejad committed:
wp delete modal changes: remove unnecessary loading and change op-icon to a span

bsatarnejad committed:
Merge branch 'redesign-of-modal-dialogues' of github.com:opf/openproject into redesign-of-modal-dialogues

bsatarnejad committed:
remove unused components

Kharonus commented on frontend/src/app/features/invite-user-modal/principal/principal.component.ts:
```diff
@@ -89,38 +116,54 @@ export class PrincipalComponent implements OnInit {
 
   get customFields():{ [key:string]:any } {
-    return this.dynamicFieldsControl?.value;
+    return this.dynamicFieldsControl?.value as { [key:string]:any };
```
Just a minor hint: if you revert this change, line 148 will still has a linter error, but your PR would be green, as you didn't changed this line.

bsatarnejad committed:
use spot-list for add-widget modal

bsatarnejad committed:
wp table configuration modal changes

bsatarnejad committed:
Merge branch 'dev' into redesign-of-modal-dialogues

bsatarnejad committed:
fix finding list item in add widget modal specs

Kharonus committed:
Chore/modal spacing (#11314)

* [chore] reworked spot-container spacings

* [chore] applied spot-container to spot-modal--body

- added style override for spacing within modal body

* [chore] removed unused modal footer class

* Remove scrollbar from project-include

Project include was creating scrollbars for the project list even if
it was not needed. This was caused by tooltips that were hidden in the
background but still affecting layouting.

This commit hides the tooltips correctly, and aligns the HTML structure
of project include to match that of file link list. A minimum height is
added to the project include list to make sure tooltips are readable even
if there is only one item in the list.

* [chore] rework of datepicker modal

- changed spacings, removed inner margins
- changed anchors to buttons (spot-link)
- added flatpickr override
- removed min-height
- moved project include min-height to root element

* [chore] fixed test selector

Co-authored-by: Benjamin Bädorf <b.baedorf@openproject.com>

bsatarnejad committed:
redesign the save query modal changes

bsatarnejad committed:
remove op-form class from invite user modal since we have spot-container now

bsatarnejad committed:
add text and title for ee modal deletion

bsatarnejad committed:
showing the error message when it is not valid value in invite user modal

bsatarnejad committed:
close the inroduction video modal  with button

bsatarnejad committed:
close the dynamic content modal without jquery

bsatarnejad committed:
fx eslint warnnings

bsatarnejad committed:
webhook response modal

Kharonus commented on config/locales/js-en.yml:
```diff
@@ -1131,12 +1131,14 @@ en:
       destroy_work_package:
         title: "Confirm deletion of %{label}"
-        text: "Are you sure you want to delete the following %{label} ?"
+        single_text: "Are you sure you want to delete the work package"
```
minor: ist there a `?` missing?

bsatarnejad replied:
No this text will be followed by the WP subject and ID, and then will be a question mark in html.

Kharonus replied:
great

Kharonus commented on config/locales/js-en.yml:
```diff
@@ -1274,11 +1278,12 @@ en:
 
       summary:
         next_button: 'Send invitation'
+        cancel_button: 'Cancel'
```
why having multiple `cancel` locale keys? we have the `js.button_cancel` key, which IMHO can be used for all of those above.

Kharonus commented on frontend/src/app/features/invite-user-modal/project-selection/project-selection.component.html:
```diff
@@ -62,9 +62,17 @@
+      (click)="close.emit()"
+    >
+    {{ text.cancelButton }}
+  </button>
```
minor: indention is off

b12f reviewed (approved):
LGTM, awesome work

Kharonus commented on frontend/src/app/shared/components/modals/save-modal/save-query.modal.html:
```diff
@@ -1,16 +1,16 @@
 >
-  <div class="spot-modal--header">{{text.save_as}}</div>
-
+  <h1 class="spot-modal--header">{{text.save_as}}</h1>
```
usually the header doesn't need to be `h1`. Why do we use it here instead of plain `div`?

b12f replied:
Was my idea; it is semantically correct HTML. The advantage we have with the typography spot classes is that you can write HTML that is semantically correct (this is the main heading of the modal, thus `h1`) but is easily made to look correct in context

Kharonus replied:
got it, no change needed then. I just feared, this might drag some unwanted margins in.

Kharonus commented on frontend/src/app/spot/styles/sass/components/modal.sass:
```diff
@@ -38,7 +38,10 @@
 
-    .spot-icon
+    + .spot-divider
+      margin-top: $spot-spacing-1
```
this should not be needed? the top margin of the divider is defined in `divider.sass`

b12f replied:
We [removed this](https://github.com/opf/openproject/pull/11246/files#diff-64ce798fd9b6656777121aa833252cfe12e3d84a026c30eb47461ad38f88cc85L2) because it breaks the BEM assumption that blocks never have margins.

Instead we added them to the context right here (spot-modal) and [in spot-container](https://github.com/opf/openproject/pull/11246/files#diff-2246b4e29d74b4ce0bee934476e29eaed67eb2c59c08bdec4e8a1ce4e2f527d5R13).

b12f replied:
I actually think `spot-divider` should maybe always be an `hr` instead of a `div`, but that's a quick change we can make any time.

Kharonus reviewed (changes requested):
I found some issues, which all should be easy fixes. Let me know, if there are any questions.

bsatarnejad committed:
remove multiple cancel translation in invite user modal

bsatarnejad committed:
fix indentation in html file

Kharonus reviewed (approved):
LGTM
